new_script.ps1

```diff
@@ -0,0 +1,172 @@
+#This script builds several branches for an Application in AppCenter and saves a report.
+#Check Readme.md file for more details - https://github.com/bumbazimba/appcenter_automatization/blob/master/README.md
+#Designed for Powershell 6 or above. Tested on Powershell 7
+#Author Nikita Nikolaev
+#Version 1.1
+
+#Enter AppCenter credentials for API requests
+
+[string]$user = Read-Host "Enter AppCenter username"
+[string]$app = Read-Host "Enter Application name"
+[string]$token = Read-Host "Enter API Token"
+[int]$branches_number = Read-Host "Enter number of branches to build"
+$branches_number = $branches_number-1
+
+
+$excel = Create-ExcelFile
+$branches = Get-Branches
+$build = Build-Branches $branches
+$build_details = Get-BuildDetails $build
+Insert-ExcelFile $excel, $build_details, $branches
+Save-ExcelFile $excel
+
+
+
+#Get All branches from the Application
+function Get-Branches ()
+{
+	$branches = Invoke-RestMethod -Uri "https://api.appcenter.ms/v0.1/apps/$user/$app/branches" -Method Get `
+		-Headers @{"Accept"="application/json"; "X-API-Token"="$($token)"; "Content-Type"="application/json"} -MaximumRetryCount 5 -RetryIntervalSec 2;
+	Write-Host "`nList of all branches in Application $app :"
+    Write-Host $branches
+    return $branches
+}
+
+
+#Build specified number of branches
+function Build-Branches ()
+{
+    param
+    (
+        [Parameter(Mandatory=$true)][object]$branches
+    )
+
+	$build = @()
+    
+	#Check if user wants to build All or just few branches
+	[int]$branches_count = $branches.Length - 1
+    $branches_names = $branches.branch.name
+        if ($branches_count -ne $branches_number)

```

Comment:
what if entered number is bigger than actual number of branches?

Reply:
Thanks for the comment, i can fix it that way: If number is bigger than maximum, it will still do maximum builds. If number is equals or lower, it will do user's number from input

    [int]$branches_count = $branches.Length - 1
        if ($branches_number -le $branches_count)
        { 
		$branches_count = $branches_number 
	}

	Foreach ($branch in $branches_names[0..$branches_count]) ...

Reply:
sounds great! thank you!